packages/expo-barcode-scanner/android/src/main/java/expo/modules/barcodescanner/BarCodeScannerViewFinder.kt

```diff
@@ -158,41 +169,47 @@ internal class BarCodeScannerViewFinder(
   override fun onPreviewFrame(data: ByteArray, innerCamera: Camera) {
     if (!barCodeScannerTaskLock) {
       barCodeScannerTaskLock = true
-      BarCodeScannerAsyncTask(innerCamera, data, barCodeScannerView).execute()
+      barCodeScannerAsyncTask(innerCamera, data)
     }
   }
 
   fun setBarCodeScannerSettings(settings: BarCodeScannerSettings?) {
     barCodeScanner.setSettings(settings)
   }
 
-  private inner class BarCodeScannerAsyncTask(private val camera: Camera?, private val mImageData: ByteArray, barCodeScannerView: BarCodeScannerView) : AsyncTask<Void?, Void?, Void?>() {
-    init {
-      this@BarCodeScannerViewFinder.barCodeScannerView = barCodeScannerView
-    }
-
-    override fun doInBackground(vararg params: Void?): Void? {
-      if (isCancelled) {
-        return null
-      }
+  private fun barCodeScannerAsyncTask(camera: Camera?, mImageData: ByteArray) {
```

Comment:
I dislike the idea of using `*AsyncTask` sufix in the method name, while it has nothing with it anymore.
```suggestion
  private fun scanForBarcodes(camera: Camera?, mImageData: ByteArray) {
```